PowerEditor/src/WinControls/ProjectPanel/ProjectPanel.cpp

```diff
@@ -595,7 +600,12 @@ void ProjectPanel::openSelectFile()
 
 void ProjectPanel::notified(LPNMHDR notification)
 {
-	if ((notification->hwndFrom == _treeView.getHSelf()))
+	if (notification->code == DMN_CLOSE)
+	{
+		::SendMessage(_hParent, WM_COMMAND, IDM_VIEW_PROJECT_PANEL_1 + _panelID, 0);
+		SetWindowLongPtr (getHSelf(), DWLP_MSGRESULT, _isClosed ? 0 : 1);
+	}
+	else if ((notification->hwndFrom == _treeView.getHSelf()))
 	{

```

Comment:
The **DWLP_MSGRESULT** mechanism is available in the _DockingManager_ and allows to prevent a tab from being closed. This mechanism gets triggered here if we press the close icon of the side window containing one or more _Project Panels_ or other docking windows. The _Project Panel_ now sends the **IDM_VIEW_PROJECT_PANEL_n** command to close itself, like the FileBrowser already does. If the close failed because the Save-Request has been cancelled, then **DWLP_MSGRESULT** is set to **1**, which prevents the tab from closing.